pkg/updatechecker/updatechecker.go

```diff
@@ -322,189 +271,209 @@ type DownloadUpdateOptions struct {
 	ClusterID      string
 	Update         upstreamtypes.Update
 	SkipPreflights bool
-	Deploy         bool
-	IsCLI          bool
 }
 
 func downloadUpdate(opts DownloadUpdateOptions) error {
-	baseArchiveDir, err := GetBaseArchiveDirForVersion(opts.AppID, opts.ClusterID, opts.Update.VersionLabel)
+	baseArchiveDir, err := version.GetBaseArchiveDirForVersion(opts.AppID, opts.ClusterID, opts.Update.VersionLabel)
 	if err != nil {
 		return errors.Wrapf(err, "failed to get base archive dir for version %s", opts.Update.VersionLabel)
 	}
 	defer os.RemoveAll(baseArchiveDir)
 
-	sequence, err := upstream.DownloadUpdate(opts.AppID, baseArchiveDir, opts.Update.Cursor, opts.SkipPreflights)
+	_, err = upstream.DownloadUpdate(opts.AppID, baseArchiveDir, opts.Update.Cursor, opts.SkipPreflights)
 	if err != nil {
 		return errors.Wrap(err, "failed to download update")
 	}
 
-	if !opts.Deploy {
-		return nil
-	}
-
-	status, err := store.GetStore().GetStatusForVersion(opts.AppID, opts.ClusterID, sequence)
-	if err != nil {
-		return errors.Wrap(err, "failed to get update downstream status")
-	}
+	return nil
+}
 
-	if status == storetypes.VersionPendingConfig {
-		logger.Infof("not deploying version %d because it's %s", sequence, status)
+func ensureDesiredVersionIsDeployed(opts CheckForUpdatesOpts, clusterID string) error {
+	if opts.DeployLatest {
+		if err := deployLatestVersion(opts, clusterID); err != nil {
+			return errors.Wrap(err, "failed to deploy latest version")
+		}
 		return nil
 	}
 
-	if err := version.DeployVersion(opts.AppID, sequence); err != nil {
-		logger.Error(errors.Wrap(err, "failed to queue update for deployment"))
+	if opts.DeployVersionLabel != "" {
+		if err := deployVersionLabel(opts, clusterID, opts.DeployVersionLabel); err != nil {
+			return errors.Wrapf(err, "failed to deploy version label %s", opts.DeployVersionLabel)
+		}
+		return nil
 	}
 
-	// preflights reporting
-	go func() {
-		err = reporting.ReportAppInfo(opts.AppID, sequence, opts.SkipPreflights, opts.IsCLI)
+	if opts.IsAutomatic {
```

Comment:
this is always true I think

Reply:
not when you check for updates manually

Reply:
https://github.com/replicatedhq/kots/pull/2333/files#diff-43223b6dda7aafc2d70ba222243d37325e88c939eac603a79f33c1042e1e9272R48